src/parser.ts

```diff
@@ -3580,10 +3579,6 @@ function parseExportDeclaration(
           tokenLine,
           tokenColumn
         );
-        if (scope) {
-          key = declaration.id ? declaration.id.name : '';
-          declareUnboundVariable(parser, key);
-        }
```

Comment:
parseFunctionDeclaration already does declareUnboundVariable. This is same for non-async function export in previous lines.